src/azure-cli-testsdk/azure/cli/testsdk/base.py

```diff
@@ -46,128 +46,49 @@ def find_recording_dir(test_file):
-
-        test_file_path = inspect.getfile(self.__class__)
-        recordings_dir = find_recording_dir(test_file_path)
-        live_test = os.environ.get(ENV_LIVE_TEST, None) == 'True'
-
-        self.vcr = vcr.VCR(
-            cassette_library_dir=recordings_dir,
-            before_record_request=self._process_request_recording,
-            before_record_response=self._process_response_recording,
-            decode_compressed_response=True,
-            record_mode='once' if not live_test else 'all',
-            filter_headers=self.FILTER_HEADERS
+        kwargs.setdefault(
+            'recording_processors',
+            [
+                SubscriptionRecordingProcessor(MOCKED_SUBSCRIPTION_ID),
+                OAuthRequestResponsesFilter(),
+                LargeRequestBodyProcessor(),
+                LargeResponseBodyProcessor(),
+                DeploymentNameReplacer(),
+                self.name_replacer,
+            ]
         )
-        self.vcr.register_matcher('query', self._custom_request_query_matcher)
-
-        self.recording_file = os.path.join(recordings_dir, '{}.yaml'.format(method_name))
-        if live_test and os.path.exists(self.recording_file):
-            os.remove(self.recording_file)
-
-        self.in_recording = live_test or not os.path.exists(self.recording_file)
-        self.test_resources_count = 0
-        self.original_env = os.environ.copy()
-
-    def setUp(self):
-        super(ScenarioTest, self).setUp()
-
-        # set up cassette
-        cm = self.vcr.use_cassette(self.recording_file)
-        self.cassette = cm.__enter__()
-        self.addCleanup(cm.__exit__)
-
-        # set up mock patches
-        patch_main_exception_handler(self)
-
-        if not self.in_recording:
-            patch_time_sleep_api(self)
-            patch_long_run_operation_delay(self)
-            patch_load_cached_subscriptions(self)
-            patch_retrieve_token_for_user(self)
-            patch_progress_controller(self)
-
-    def tearDown(self):
-        os.environ = self.original_env
+        kwargs.setdefault(
+            'replay_processors',
+            [
+                LargeResponseBodyReplacer(),
+                DeploymentNameReplacer(),
+            ]
+        )
+        kwargs.setdefault(
+            'recording_patches',
+            [
+                patch_main_exception_handler,
+            ]
+        )
+        kwargs.setdefault(
+            'replay_patches',
+            [
+                patch_main_exception_handler,
+                patch_time_sleep_api,
+                patch_long_run_operation_delay,
+                patch_load_cached_subscriptions,
+                patch_retrieve_token_for_user,
+            ]
+        )
+        kwargs.setdefault(
+            'recording_dir',
+            find_recording_dir(inspect.getfile(self.__class__))
+        )
+        super(ScenarioTest, self).__init__(method_name, **kwargs)
```

Comment:
Agreed, signature should be:
```python
def __init__(self, method_name, recording_processors, replay_processors, recording_patches, replay_patches, recording_dir):
```
but this makes a lot of parameters :/

I was wondering if at this point we should just not take a "vcr" parameter:
```
        vcr = vcr.VCR(
            cassette_library_dir=recording_dir,
            before_record_request=self._process_request_recording,
            before_record_response=self._process_response_recording,
            decode_compressed_response=True,
            record_mode=self.config.record_mode,
            filter_headers=self.FILTER_HEADERS
        )
```
and then
```python
def __init__(self, method_name, vcr, recording_patches, replay_patches):
```

This removes the need for "recording_dir", "recording_processors" and "replay_processors" and even simplify the SDK about "record_mode".

Thoughts @v-iam @troydai ?